src/utils.jl

```diff
@@ -39,4 +39,40 @@ function check_isfinite(x::Vector)
     if !isempty(i)
         throw(IsFiniteException(i))
     end
-end
\ No newline at end of file
+end
+
+
+function n_ary(f::Function)
+    f!(x::Vector, fx::AbstractArray) = copy!(fx, [f(x...)... ])
+end
+
+function not_in_place(f::Function)
+    f!(x::Vector, y::AbstractArray) = copy!(y, f(x))
+end
+
+function not_in_place(f::Function, g::Function)
+    depwarn("nlsolve(not_in_place(f, g)) is deprecated. Use nlsolve(not_in_place(f), not_in_place(g))")
```

Comment:
not exactly because one the replacement only works within nlsolve but I did change the function a little bit.